local-volume/bootstrapper/bootstrapper.go

```diff
@@ -0,0 +1,246 @@
+		},
+		Subjects: subjects,
+	}
+	nodeBinding := rbacv1beta1.ClusterRoleBinding{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "rbac.authorization.k8s.io/v1beta1",
+			Kind:       "ClusterRoleBinding",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ProvisionerNodeBindingName,
+		},
+		RoleRef: rbacv1beta1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     SystemRoleNode,
+		},
+		Subjects: subjects,
+	}
+
+	_, err := client.RbacV1beta1().ClusterRoleBindings().Create(&pvBinding)
+	if err != nil {
+		return err
+	}
+	_, err = client.RbacV1beta1().ClusterRoleBindings().Create(&nodeBinding)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func createDaemonSet(client *kubernetes.Clientset, namespace string, config map[string]common.MountConfig) error {
+	volumes := []v1.Volume{}
+	volumeMounts := []v1.VolumeMount{}
+	for _, mount := range config {
+		name := generateMountName(mount.HostDir, mount.MountDir)
+		volumes = append(volumes, v1.Volume{
+			Name: name,
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: mount.HostDir,
+				},
+			},
+		})
+		volumeMounts = append(volumeMounts, v1.VolumeMount{
+			Name:      name,
+			MountPath: mount.MountDir,
+		})
+	}
+
+	envVars := []v1.EnvVar{
+		{
+			Name: "MY_NODE_NAME",
+			ValueFrom: &v1.EnvVarSource{
+				FieldRef: &v1.ObjectFieldSelector{
+					FieldPath: "spec.nodeName",
+				},
+			},
+		},
+	}
+
+	containers := []v1.Container{
+		{
+			Name:         ProvisionerContainerName,
+			Image:        DefaultImage,
+			VolumeMounts: volumeMounts,
+			Env:          envVars,
+		},
+	}
+
+	daemonSet := extv1beta1.DaemonSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "extensions/v1beta1",
+			Kind:       "DaemonSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ProvisionerDaemonSetName,
+			Namespace: namespace,
+		},
+		Spec: extv1beta1.DaemonSetSpec{
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"app": ProvisionerDaemonSetName},
+				},
+				Spec: v1.PodSpec{
+					Volumes:            volumes,
+					Containers:         containers,
+					ServiceAccountName: ProvisionerServiceAccountName,
+				},
+			},
+		},
+	}
+
+	_, err := client.Extensions().DaemonSets(namespace).Create(&daemonSet)
+	return err
+}
+
+func main() {
+	flag.Set("logtostderr", "true")
+	flag.Parse()
+
+	namespace := os.Getenv("MY_NAMESPACE")
+	if namespace == "" {
+		glog.Fatalf("MY_NAMESPACE environment variable not set\n")
+	}
+	volumeConfigName := os.Getenv("VOLUME_CONFIG_NAME")
+	if volumeConfigName == "" {
+		glog.Fatalf("VOLUME_CONFIG_NAME environment variable not set\n")
+	}
+
+	client := setupClient()
+	config := getVolumeConfig(client, namespace, volumeConfigName)
+	glog.Infof("Running bootstrap pod with config %+v\n", config)
+
+	// TODO: check error and clean up resources.
+	if err := createServiceAccount(client, namespace); err != nil {
```

Comment:
updated to not fail.

yeah, it's left as a TODO to make bootstrapper configurable.
> Would the admin want to provide their own accounts, instead of the bootstrapper creating it for them?